File: 프로그래머스/2/12924. 숫자의 표현/숫자의 표현.py
def solution(n):
    answer = 1
    left, right = 1, 1
    total = 1
    
    while left <= n // 2:
        if total == n:
            answer += 1
        
        if total >= n:
            total -= left
            left += 1
        else:
            right += 1
            total += right
    
    return answer

File: 프로그래머스/2/12924. 숫자의 표현/숫자의 표현.cpp
#include <string>
#include <vector>
#include <iostream>
#include <algorithm>

using namespace std;

int solution(int n) {
    int answer = 1;
    
    int sum = 1;
    int left = 1;
    int right = 1;
    
    while (left <= n / 2) {
        if (sum == n) answer++;
        
        if (sum >= n) {
            sum -= left;
            left++;
        }
        else {
            right++;
            sum += right;
        }
    }
    
    return answer;
}